test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs

```diff
@@ -263,5 +263,79 @@ public void IncompatibleSourcesWarningShouldBeDisplayedInTheConsoleOnlyWhenRunni
                 this.StdOutputContains(expectedWarningContains);
             }
         }
+
+        [TestMethod]
+        [NetFullTargetFrameworkDataSource]
+        public void ExitCodeShouldReturnOneWhenFailWhenNoTestsFoundParameterSetToTrueAndNoTestAvailableWithFilter(RunnerInfo runnerInfo)
+        {
+            SetTestEnvironment(this.testEnvironment, runnerInfo);
+
+            var assemblyPaths = this.BuildMultipleAssemblyPath("SimpleTestProject2.dll").Trim('\"');
+
+            var arguments = PrepareArguments(assemblyPaths, this.GetTestAdapterPath(), string.Empty, this.FrameworkArgValue, runnerInfo.InIsolationValue);
+            
+            // Setting /TestCaseFilter to random test name, which does not exists
+            arguments = string.Concat(arguments, " /TestCaseFilter:SomeRandomTestName");
+
+            // set FailWhenNoTestsFound to true
+            arguments = string.Concat(arguments, " -- RunConfiguration.FailWhenNoTestsFound=true");
+            this.InvokeVsTest(arguments);
+
+            this.ExitCodeEquals(1);
+        }
+
+        [TestMethod]
+        [NetFullTargetFrameworkDataSource]
+        public void ExitCodeShouldReturnZeroWhenFailWhenNoTestsFoundParameterSetToTrueAndNoTestAvailableWithFilter(RunnerInfo runnerInfo)
+        {
+            SetTestEnvironment(this.testEnvironment, runnerInfo);
+
+            var assemblyPaths = this.BuildMultipleAssemblyPath("SimpleTestProject2.dll").Trim('\"');
+
+            var arguments = PrepareArguments(assemblyPaths, this.GetTestAdapterPath(), string.Empty, this.FrameworkArgValue, runnerInfo.InIsolationValue);
+            
+            // Setting /TestCaseFilter to random test name, which does not exists
+            arguments = string.Concat(arguments, " /TestCaseFilter:SomeRandomTestName");
+
+            // set FailWhenNoTestsFound to true
+            arguments = string.Concat(arguments, " -- RunConfiguration.FailWhenNoTestsFound=false");
+            this.InvokeVsTest(arguments);
+
+            this.ExitCodeEquals(0);
+        }
+
+        [TestMethod]
+        [NetFullTargetFrameworkDataSource]
+        public void ExitCodeShouldNotDependOnFailWhenNoTestsFoundTrueValueWhenTestsAreAvailable(RunnerInfo runnerInfo)
+        {
+            SetTestEnvironment(this.testEnvironment, runnerInfo);
+
+            var assemblyPaths = this.BuildMultipleAssemblyPath("SimpleTestProject2.dll").Trim('\"');
+
+            var arguments = PrepareArguments(assemblyPaths, this.GetTestAdapterPath(), string.Empty, this.FrameworkArgValue, runnerInfo.InIsolationValue);
+
+            // set FailWhenNoTestsFound to true
+            arguments = string.Concat(arguments, " -- RunConfiguration.FailWhenNoTestsFound=true");
+            this.InvokeVsTest(arguments);
+
+            this.ExitCodeEquals(1);
```

Comment:
Add a comment that this is because there is 1 failing test in the assembly, if that is the reason. If that is not the reason it is not obvious why this is 1 and not easy to differntiate it from the other types of failure that we are testing here. 

Same comment would be nice in the last test as well, because we prove the same thing there just with the option set to the opposite value.